src/pip/_internal/utils/entrypoints.py

```diff
@@ -1,7 +1,23 @@
+import itertools
+import os
+import shutil
 import sys
 from typing import List, Optional
 
 from pip._internal.cli.main import main
+from pip._internal.utils.compat import WINDOWS
+
+_EXECUTABLE_NAMES = [
+    "pip",
+    f"pip{sys.version_info.major}",
+    f"pip{sys.version_info.major}.{sys.version_info.minor}",
+]
+if WINDOWS:
+    _allowed_extensions = {"", ".exe", ".EXE"}
```

Comment:
We really should be doing case insensitive comparisons rather than trying to guess the case of the extension here. But 🤷 I guess the worst that happens is we don't suggest a shorter form when we could...

Reply:
You're correct!